src/app.js

```diff
@@ -1,21 +1,59 @@
 'use strict';
 
-/**
- * Implement sum function:
- *
- * Function takes 2 numbers and returns their sum
- *
- * sum(1, 2) === 3
- * sum(1, 11) === 12
- *
- * @param {number} a
- * @param {number} b
- *
- * @return {number}
- */
-function sum(a, b) {
-  // write code here
-  return a + b;
-}
-
-module.exports = sum;
+const http = require('http');
+const path = require('path');
+const fs = require('fs/promises');
+
+const server = () => http.createServer(async(req, res) => {
+  const normalizedUrl = new URL(req.url, `http://${req.headers.host}`);
+  const pathname = normalizedUrl.pathname;
+  const fileName = path.basename(pathname);
+  const dirName = path.dirname(pathname);
+  const fileExt = path.extname(pathname);
+
+  if (fileName === 'file') {
+    res.setHeader('Content-Type', 'text/html');
+
+    try {
+      const data = await fs.readFile('./public/index.html');
+
+      res.statusCode = 200;
+      res.statusText = 'OK';
+      res.end(data);
+    } catch (err) {
+      res.statusCode = 404;
+      res.statusText = 'Not Found';
+      res.end();
+    }
+
+    return;
+  }
+
+  if (dirName.startsWith('/file')) {
+    res.setHeader('Content-Type', `text/${fileExt.slice(1)}`);
+
+    const normalizedDirname = dirName.replace('/file', '/public');
+
+    try {
+      const data = await fs.readFile(`.${normalizedDirname}/${fileName}`);
```

Comment:
The path is calculated wrong, it doesn't work for me. Please use helpers in `path` module